### PR TITLE
Regex concept

### LINKED ISSUE #122: Regular Expressions concept
- [x] select an exercise
- [x] make a first draft of the design.md file, laying out the goals for the concept & exercise.
- [x] add concept and exercise to the "root" config.json
- [x] edit concept's and exercise's config.json
- [x] write tests, exemplar and stub
- [x] edit instructions
- [x] revisit design.md
- [x] write about.md
- [x] copy and prune to create introduction.md
- [x] edit hints
- [x] add links

### PR DESCRIPTION
**Pull request by glennj**
## Summary

Create the Regular Expressions concept.
Create the "Regular Chatbot" exercise.

Closes #122 

## Checklist
- [x] If docs where changed, run `./bin/configlet generate` to ensure all documents are properly generated.
- [x] CI is green

### CODE REVIEW

File: exercises/concept/regular-chatbot/.meta/design.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+- How to use the regex related functions
+    - test, match, capture, split, scan, sub, gsub
+- How to pass flags
+- Understanding the different parts of the match result, incl. capture results
+- Using the replace method alongside regular expressions
+
+## Out of scope
+
+This exercise is intended as an introduction to Regular Expressions in jq.
+The main focus resides in understanding the different functions to **use** regexes
```
Comment (IsaacG) [inline exercises/concept/regular-chatbot/.meta/design.md:16]:
```suggestion
The main focus is understanding the different functions to **use** regexes.
```

File: concepts/regular-expressions/about.md [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+That means that any backslashes in the regular expression need to be escaped in the string.
+
+For example, the digit character class (`\d`) must be written like `"\\d"`.
+~~~~
+
+## Regex Functions
+
+Regular expressions in `jq` are limited to [a set of filters][jq-regex-funcs].
+
+### Simple matching
```
[the remark this answers is not in the document]
Comment (glennj) [reply concepts/regular-expressions/about.md:27]: Will reconcile.

File: exercises/concept/regular-chatbot/.docs/instructions.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+## Get Website Link
+
+The Chatbot is a really curious software.
+Even though it can search on the internet about a particular topic, it likes to ask users about cool websites or URLs to go find relevant information.
+
+Example of Conversation:
+
+> **Chatbot**: Hey User, I would like to learn how to code in JavaScript, do you know any cool website where I could learn?
+
+> **User**: I learned a lot from exercism.org, there's lots of great stuff there.
```
[the remark this answers is not in the document]
Comment (glennj) [reply exercises/concept/regular-chatbot/.docs/instructions.md:68]: If you've got it, flaunt it.

File: exercises/concept/remote-control-car/.docs/introduction.md [hunk 4]
```diff
@@ -57,7 +57,7 @@ A _function_ introduces a new **scope** for variables and nested functons.
 
 ### Arguments
 
-_Function_ arguments are separated by _semi-colons_ not commas.
```
[the remark this answers is not in the document]
Comment (glennj) [reply exercises/concept/remote-control-car/.docs/introduction.md:60]: `configlet generate` doesn't have the ability to limit to just one exercise. It's a bit dirty to leave them here, but I didn't take the effort to undo it.

File: exercises/concept/regular-chatbot/.docs/instructions.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
+# Instructions
+
+You have been hired as a Regular Expression Specialist in a company that is developing a Chatbot.
+
+It is in a very basic phase of development, hence your mission is to use Regular Expressions to improve the chatbot's ability to understand and generate natural language.
+
+## Check Valid Command
+
+Apart from being smart, the Chatbot is also a loyal assistant.
```
[the remark this answers is not in the document]
Comment (glennj) [reply exercises/concept/regular-chatbot/.docs/instructions.md:10]: will fix

File: exercises/concept/regular-chatbot/.docs/instructions.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+Implement the `check_phone_number` function.
+
+If the number is valid, the chatbot answers with a message thanking the user and confirming the number.
+If the number is invalid, the function informs the user that the phone number is not valid.
+
+The expected format is `(+NN) NNN-NNN-NNN`, where N is a digit.
+
+```jq
+"(+34) 659-771-594" | check_phone_number
+# => "Thanks! You can now download me to your phone."
```
[the remark this answers is not in the document]
Comment (glennj) [reply exercises/concept/regular-chatbot/.docs/instructions.md:56]: clarifying

File: exercises/concept/regular-chatbot/.docs/instructions.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+Just remove the emoji string. Do not attempt to adjust the whitespace.
+
+```jq
+"I love playing videogames emoji3465 it's one of my hobbies" | remove_emoji
+# => "I love playing videogames  it's one of my hobbies"
+```
+
+## Check Valid Phone Number
+
+Considering the download of chatbot features on a mobile app, the user is expected to write a phone number during the conversation.
```
[the remark this answers is not in the document]
Comment (glennj) [reply exercises/concept/regular-chatbot/.docs/instructions.md:44]: rewriting

File: exercises/concept/regular-chatbot/regular-chatbot.jq [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+#   where N is a digit
+#
+# input: {string} number
+# output: {string} the Chatbot response to the phone Validation
+def check_phone_number:
+  . # implement the body of this function
+;
+
+# Given a certain response from the user, help the chatbot get
+# only the URLs
```
[the remark this answers is not in the document]
Comment (glennj) [reply exercises/concept/regular-chatbot/regular-chatbot.jq:34]: domains. updating.

**Comment by glennj** [comment]
The main question: does the concept and exercise satisfy the design goals?

**Comment by IsaacG** [comment]
> The main question: does the concept and exercise satisfy the design goals?

I believe it does!

**Review by IsaacG — APPROVED** [review]

**Pull request merged**